test-data/unit/check-modules.test

```diff
@@ -1943,3 +1943,24 @@ main:2: error: Name 'name' is not defined
 main:3: error: Revealed type is 'Any'
 
 [builtins fixtures/module.pyi]
+
+
+[case testImportCycle]
+from m import One
+reveal_type(One)
+[file m/__init__.py]
+from .one import One
+from .two import Two
+reveal_type(One)
+[file m/one.py]
+class One:
+    pass
+[file m/two.py]
+from m import One
+reveal_type(One)
+class Two:
+    pass
+[out]
```

Comment:
Add also tests for incremental mode.

Reply:
Ok, I can look at some existing incremental-mode tests to see how that's done.